Core/PrivacyPractices.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  PrivacyPractices.swift
+//  DuckDuckGo
+//
+//  Copyright © 2017 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// The main interface for privacy practices.  Currently uses TOSDR as its data source.
+public class PrivacyPractices {
+    
+    public enum Summary {
+        case poor, mixed, good, unknown
+    }
+    
+    public struct Practice {
+        
+        public let score: Int
+        public let summary: Summary
+        public let goodReasons: [String]
+        public let badReasons: [String]
+        
+    }
+    
+    struct Constants {
+        static let unknown = Practice(score: 2, summary: .unknown, goodReasons: [], badReasons: [])
+    }
+
+    private let tld: TLD
+    private let entityScores: [String: Int]
+    private let siteScores: [String: Int]
+    private let termsOfServiceStore: TermsOfServiceStore
+    private let entityMapping: EntityMapping
+    
+    public init(termsOfServiceStore: TermsOfServiceStore = EmbeddedTermsOfServiceStore(), entityMapping: EntityMapping = EntityMapping()) {
+        let tld = TLD()
+        var entityScores = [String: Int]()
+        var siteScores = [String: Int]()
+        
+        termsOfServiceStore.terms.forEach {
```

Comment:
I've made TOSDR the underlying implementation for the privacy practices.  This constructor floats the score up to the parent, but the score is actually derived as a function on the `TermsOfService` struct.